src/utils/getEndpoint.ts

```diff
@@ -7,10 +7,11 @@ import { isFilePathString } from './guards.js';
 
 export default function getEndpoint(
   stage: Stage,
-  type: 'iterator' | 'generator' = 'iterator'
+  type: 'iterator' | 'generator' = 'iterator',
+  index?: number
 ): Endpoint {
   const t: keyof LDWorkbenchConfiguration['stages'][0] = type
-  const endpoint = stage.configuration[t].endpoint
+  const endpoint = t === "generator" ? stage.configuration[t]?.[index!]?.endpoint : stage.configuration[t]?.endpoint;
```

Comment:
This reads like `index` is a required argument iff `type === "generator"`, but this isn't reflected in the call-signature.

I would make the following call signature:

```ts
export default function getEndpoint(
  stage: Stage,
  type?: 'iterator'
): Endpoint;
export default function getEndpoint(
  stage: Stage,
  type: 'generator',
  index: number
): Endpoint;
export default function getEndpoint(
  stage: Stage,
  type: 'iterator' | 'generator' = 'iterator',
  index?: number
): Endpoint {
```

This allows for the following calling options:

```ts
getEndpoint(stage) // defaults to `iterator`
getEndpoint(stage, "iterator")
getEndpoint(stage, "generator", 0)
```

But disallows the following:
```ts
getEndpoint(stage, "generator") // this would have caused a runtime issue in the current situation
const t = "iterator" as "iterator" | "generator"
getEndpoint(stage, t) // this may be incorrect, since the compiler can't know whether `t === "generator"`, and it therefore correctly reports an error
getEndpoint(stage, t, 1) // this is also a compile error, even though it never leads to issues, but it's a bit weird, since the index is ignored if `t === "iterator"`. If you'd like to allow this, add `| 'iterator'` in the second call signature
```